x-pack/plugins/case/server/client/client.ts

```diff
@@ -59,96 +61,190 @@ export class CaseClientHandler implements CaseClient {
     this._savedObjectsClient = clientArgs.savedObjectsClient;
     this._userActionService = clientArgs.userActionService;
     this._alertsService = clientArgs.alertsService;
+    this.logger = clientArgs.logger;
   }
 
   public async create(caseInfo: CasePostRequest) {
-    return create({
-      savedObjectsClient: this._savedObjectsClient,
-      caseService: this._caseService,
-      caseConfigureService: this._caseConfigureService,
-      userActionService: this._userActionService,
-      user: this.user,
-      theCase: caseInfo,
-    });
+    try {
+      return create({
+        savedObjectsClient: this._savedObjectsClient,
+        caseService: this._caseService,
+        caseConfigureService: this._caseConfigureService,
+        userActionService: this._userActionService,
+        user: this.user,
+        theCase: caseInfo,
+        logger: this.logger,
+      });
+    } catch (error) {
+      throw createCaseError({
+        message: `Failed to create a new case using client: ${error}`,
+        error,
+        logger: this.logger,
+      });
+    }
   }
 
   public async update(cases: CasesPatchRequest) {
-    return update({
-      savedObjectsClient: this._savedObjectsClient,
-      caseService: this._caseService,
-      userActionService: this._userActionService,
-      user: this.user,
-      cases,
-      caseClient: this,
-    });
+    try {
+      return update({
+        savedObjectsClient: this._savedObjectsClient,
+        caseService: this._caseService,
+        userActionService: this._userActionService,
+        user: this.user,
+        cases,
+        caseClient: this,
+        logger: this.logger,
+      });
+    } catch (error) {
+      const caseIDVersions = cases.cases.map((caseInfo) => ({
+        id: caseInfo.id,
+        version: caseInfo.version,
+      }));
+      throw createCaseError({
```

Comment:
I initially thought the `createCaseError` identified errors for the `createCase` path, but it looks like a catch all for all case related errors. Think it's worth changing the name to just `caseError` or `logCaseErrors` or something?

Reply:
Yeah good idea. I'll switch it to `caseError` 👍